File: backend/parse_supervisors.php
<?php
$data = array_map('str_getcsv', file('storage/app/supervisors.csv'));
$supervisors = [];

foreach ($data as $row) {
    $name = trim($row[1] ?? '');
    $email = trim($row[2] ?? '');
    $phone = trim($row[3] ?? '');
    
    if (empty($name) || empty($email) || strpos($email, '@') === false || strtolower($email) === 'e-mail') {
        continue;
    }
    
    // Clean name: remove trailing spaces, standardize
    $supervisors[] = [
        'name' => $name,
        'email' => $email,
        'phone' => $phone
    ];
}

file_put_contents('storage/app/supervisors.json', json_encode($supervisors, JSON_PRETTY_PRINT));
echo "Found " . count($supervisors) . " supervisors.\n";

File: backend/parse_supervisors.py
import pandas as pd
import json

df = pd.read_csv('storage/app/supervisors.csv')
supervisors = []

for index, row in df.iterrows():
    name = str(row['Unnamed: 1']).strip()
    email = str(row['Unnamed: 2']).strip()
    phone = str(row['Unnamed: 3']).strip()
    
    if name == 'nan' or name == 'Supervisor Name' or name == '' or email == 'nan' or email == 'E-mail':
        continue
        
    if '@' in email:
        supervisors.append({
            'name': name,
            'email': email,
            'phone': phone if phone != 'nan' else ''
        })

with open('storage/app/supervisors.json', 'w') as f:
    json.dump(supervisors, f, indent=4)
